README.md
# flutter_gradient_colors

A new Flutter package project.

# Show some :heart: and star the repo to support the project

# 100 New Colors Added

| # ScreenShots                       | # Screenshot                        | # ScreenShots                       |
| ----------------------------------- | ----------------------------------- | ----------------------------------- |
| ![](./screenshots/newColors/1.jpg)  | ![](./screenshots/newColors/2.jpg)  | ![](./screenshots/newColors/3.jpg)  |
| ![](./screenshots/newColors/4.jpg)  | ![](./screenshots/newColors/5.jpg)  | ![](./screenshots/newColors/6.jpg)  |
| ![](./screenshots/newColors/7.jpg)  | ![](./screenshots/newColors/8.jpg)  | ![](./screenshots/newColors/9.jpg)  |
| ![](./screenshots/newColors/10.jpg) | ![](./screenshots/newColors/11.jpg) | ![](./screenshots/newColors/12.jpg) |
| ![](./screenshots/newColors/13.jpg) | ![](./screenshots/newColors/14.jpg) | ![](./screenshots/newColors/15.jpg) |
| ![](./screenshots/newColors/16.jpg) | ![](./screenshots/newColors/17.jpg) | ![](./screenshots/newColors/18.jpg) |
| ![](./screenshots/newColors/19.jpg) | ![](./screenshots/newColors/20.jpg) | ![](./screenshots/newColors/21.jpg) |
| ![](./screenshots/newColors/22.jpg) |                                     |                                     |

| # ScreenShots                                         | # Screenshot                                          | # ScreenShots                                         |
| ----------------------------------------------------- | ----------------------------------------------------- | ----------------------------------------------------- |
| ![](./screenshots/Screenshot_2020-01-07-16-51-03.png) | ![](./screenshots/Screenshot_2020-01-07-16-51-13.png) | ![](./screenshots/Screenshot_2020-01-07-16-51-28.png) |
| ![](./screenshots/Screenshot_2020-01-07-16-51-33.png) | ![](./screenshots/Screenshot_2020-01-07-16-51-42.png) | ![](./screenshots/Screenshot_2020-01-07-16-51-51.png) |
| ![](./screenshots/Screenshot_2020-01-07-16-52-00.png) | ![](./screenshots/Screenshot_2020-01-07-16-52-10.png) | ![](./screenshots/Screenshot_2020-01-07-16-52-16.png) |
| ![](./screenshots/Screenshot_2020-01-07-16-52-23.png) | ![](./screenshots/Screenshot_2020-01-07-16-52-29.png) | ![](./screenshots/Screenshot_2020-01-07-16-52-36.png) |
| ![](./screenshots/Screenshot_2020-01-07-16-52-48.png) | ![](./screenshots/Screenshot_2020-01-07-16-52-54.png) | ![](./screenshots/Screenshot_2020-01-07-16-53-02.png) |
| ![](./screenshots/Screenshot_2020-01-07-16-53-15.png) | ![](./screenshots/Screenshot_2020-01-07-16-53-24.png) |                                                       |

# Types of Gradient Color class

```
      1. GradientColor class: In this Gradient color class we use two colors for gradient color, so use only two value for stop otherwise you will face error.
        eg. Container(
            margin: EdgeInsets.only(top: 10.0),
            width: MediaQuery.of(context).size.width - 10,
            height: 140,
            decoration: BoxDecoration(
                gradient: LinearGradient(
              begin: Alignment.topLeft,
              end: Alignment.bottomRight,
              colors: GradientColors.blue,
              stop:[
              0.6,
              0.7
              ]
            )),
          ),

      2. MoreGradientColor class: In this Gradient color class we use three colors for gradient color, so use only three value for stop otherwise you will face error.

      eg. Container(
            margin: EdgeInsets.only(top: 10.0),
            width: MediaQuery.of(context).size.width - 10,
            height: 140,
            decoration: BoxDecoration(
                gradient: LinearGradient(
              begin: Alignment.topLeft,
              end: Alignment.bottomRight,
              colors: MoreGradientColors.instagram,
              stop:[
              0.3,
              0.6,
              0.9
              ]
            )),
          ),
```

# New Colors List (GradientColors)

| # Color Name    | # Color Name    | # Color Name     |
| --------------- | --------------- | ---------------- |
| warmFlame       | nightFade       | springWarmt      |
| juicyPeach      | ladyLips        | sunnyMorning     |
| rainyAshville   | frozenDreams    | winterNeva       |
| dustyGrass      | temptingAzure   | heavyRain        |
| amyCrisp        | meanFruit       | lightBluee       |
| ripeMalinka     | cloudyKnoxville | malibuBeach      |
| newLife         | trueSunset      | morpheusDen      |
| rareWind        | wildApple       | plumPlate        |
| everlastingSky  | happyFisher     | blessingGet      |
| sharpeyeEagle   | liadogaBottom   | lemonGate        |
| itmeoBranding   | zeusMiracle     | deepBlue         |
| happyAcid       | awesomePine     | newYork          |
| mixedHopes      | flyHigh         | freshMilk        |
| snowAgain       | februaryInk     | kindSteel        |
| softGrass       | grownEarly      | sharpBlues       |
| shadyWater      | dirtyBeauty     | greatWhale       |
| teenNotebook    | politeRumors    | redSalvation     |
| nightParty      | skyGlider       | heavenPeach      |
| purpleDivision  | aquaSplash      | redLove          |
| cleanMirror     | premiumDark     | cochitiLake      |
| summerGames     | passionateRed   | mountainRock     |
| desertHump      | jungleDay       | phoenixStart     |
| octoberSilence  | farawayRiver    | alchemistLab     |
| overSun         | marsParty       | eternalConstance |
| japanBlush      | summerGames     | smilingRain      |
| cloudy          | bigMango        | healthyWater     |
| amour           | strongStick     | blackGray        |
| paloAlto        | happyMemories   | midnightBloom    |
| crystalline     | partyBliss      | confidentCloud   |
| leCocktail      | riverCity       | rozenBerry       |
| childCare       | flyingLemon     | hiddenJaguar     |
| nega            | seashore        | marbleWall       |
| cheerfulCaramel | nightSky        | youngGrass       |
| nega            | seashore        | marbleWall       |
| colorfulPeach   | gentleCare      | plumBath         |
| happyUnicorn    | glassWater      |                  |

# Colors List (GradientColors)

| # Color Name      | # Color Name   | # Color Name    |
| ----------------- | -------------- | --------------- |
| blue              | pink           | orange          |
| indigo            | white          | black           |
| piggyPink         | coolBlues      | eveningSunshine |
| darkOcean         | gradeGrey      | dimBlue         |
| ver               | lightBlue      | lightGreen      |
| mango             | royalBlue      | skyLine         |
| darkPink          | purplePink     | purplePink      |
| skyBlue           | seaBlue        | noontoDusk      |
| red               | lightBrown     | harmonicEnergy  |
| radish            | sunny          | teal            |
| purple            | green          | yellow          |
| orangePink        | aqua           | sunrise         |
| cherry            | mojito         | juicyOrange     |
| mirage            | violet         | lightBlack      |
| facebookMessenger | winter         | cloud           |
| grey              | beautifulGreen |                 |

# Colors List (MoreGradientColors)

| # Color Name   | # Color Name | # Color Name |
| -------------- | ------------ | ------------ |
| hazel          | darkSkyBlue  | instagram    |
| jShine         | coolSky      | azureLane    |
| orangePinkBlue | lunada       |              |

# Example

```
class ContainerWidget extends StatelessWidget {
  final List<Color> colors;
  final String colorName;

  const ContainerWidget(
      {Key key, @required this.colors, @required this.colorName})
      : super(key: key);
  @override
  Widget build(BuildContext context) {
    return Center(
      child: Stack(
        children: <Widget>[
          Container(
            margin: EdgeInsets.only(top: 10.0),
            width: MediaQuery.of(context).size.width - 10,
            height: 140,
            decoration: BoxDecoration(
                gradient: LinearGradient(
              begin: Alignment.topLeft,
              end: Alignment.bottomRight,
              colors: colors,
            )),
          ),
          Positioned(
            bottom: 0,
            left: 0,
            right: 0,
            child: Container(
              height: 30.0,
              child: Center(
                child: Text(
                  "$colorName",
                  style: TextStyle(color: Colors.white, fontSize: 20.0),
                ),
              ),
              color: Color.fromRGBO(0, 0, 0, 0.3),
            ),
          )
        ],
      ),
    );
  }
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: "Flutter Gradient Color",
      debugShowCheckedModeBanner: false,
      home: Scaffold(
        appBar: AppBar(
          title: Text("Flutter Gradient Color"),
        ),
        body: BodyPart(),
      ),
    );
  }
}

class BodyPart extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return CupertinoScrollbar(
      child: ListView(
        children: <Widget>[
          ContainerWidget(
            colorName: "Blue",
            colors: GradientColors.blue,
          ),
          ContainerWidget(
            colorName: "Cool Blue",
            colors: GradientColors.coolBlues,
          ),
          ContainerWidget(
            colorName: "Pink",
            colors: GradientColors.pink,
          ),
          ContainerWidget(
            colorName: "Orange",
            colors: GradientColors.orange,
          ),
          ContainerWidget(
            colorName: "Indigo",
            colors: GradientColors.indigo,
          ),
          ContainerWidget(
            colorName: "White",
            colors: GradientColors.white,
          ),
          ContainerWidget(
            colorName: "Black",
            colors: GradientColors.black,
          ),
          ContainerWidget(
            colorName: "Piggy Pink",
            colors: GradientColors.piggyPink,
          ),
          ContainerWidget(
            colorName: "Evening Sunshine",
            colors: GradientColors.eveningSunshine,
          ),
          ContainerWidget(
            colorName: "Dark Ocean",
            colors: GradientColors.darkOcean,
          ),
          ContainerWidget(
            colorName: "Grade Grey",
            colors: GradientColors.gradeGrey,
          ),
          ContainerWidget(
            colorName: "Dim Blue",
            colors: GradientColors.dimBlue,
          ),
          ContainerWidget(
            colorName: "Ver",
            colors: GradientColors.ver,
          ),
          ContainerWidget(
            colorName: "LightBlue",
            colors: GradientColors.lightBlue,
          ),
          ContainerWidget(
            colorName: "Lunada",
            colors: MoreGradientColors.lunada,
          ),
          ContainerWidget(
            colorName: "Light Green",
            colors: GradientColors.lightGreen,
          ),
          ContainerWidget(
            colorName: "Violet",
            colors: GradientColors.violet,
          ),
          ContainerWidget(
            colorName: "Light Black",
            colors: GradientColors.lightBlack,
          ),
          ContainerWidget(
            colorName: "Facebook Messenger",
            colors: GradientColors.facebookMessenger,
          ),
          ContainerWidget(
            colorName: "Winter",
            colors: GradientColors.winter,
          ),
          ContainerWidget(
            colorName: "Cloud",
            colors: GradientColors.cloud,
          ),
          ContainerWidget(
            colorName: "Grey",
            colors: GradientColors.grey,
          ),
          ContainerWidget(
            colorName: "Beautiful Green",
            colors: GradientColors.beautifulGreen,
          ),
          ContainerWidget(
            colorName: "Mango",
            colors: GradientColors.mango,
          ),
          ContainerWidget(
            colorName: "Royal Blue",
            colors: GradientColors.royalBlue,
          ),
          ContainerWidget(
            colorName: "Sky Line",
            colors: GradientColors.skyLine,
          ),
          ContainerWidget(
            colorName: "Dark Pink",
            colors: GradientColors.darkPink,
          ),
          ContainerWidget(
            colorName: "Purple Pink",
            colors: GradientColors.purplePink,
          ),
          ContainerWidget(
            colorName: "Sky Blue",
            colors: GradientColors.skyBlue,
          ),
          ContainerWidget(
            colorName: "Sea Blue",
            colors: GradientColors.seaBlue,
          ),
          ContainerWidget(
            colorName: "Hazel",
            colors: MoreGradientColors.hazel,
          ),
          ContainerWidget(
            colorName: "Noon To Dusk",
            colors: GradientColors.noontoDusk,
          ),
          ContainerWidget(
            colorName: "Red",
            colors: GradientColors.red,
          ),
          ContainerWidget(
            colorName: "Light Brown",
            colors: GradientColors.lightBrown,
          ),
          ContainerWidget(
            colorName: "Harmonic Energy",
            colors: GradientColors.harmonicEnergy,
          ),
          ContainerWidget(
            colorName: "Radish",
            colors: GradientColors.radish,
          ),
          ContainerWidget(
            colorName: "Sunny",
            colors: GradientColors.sunny,
          ),
          ContainerWidget(
            colorName: "Teal",
            colors: GradientColors.teal,
          ),
          ContainerWidget(
            colorName: "Purple",
            colors: GradientColors.purple,
          ),
          ContainerWidget(
            colorName: "Green",
            colors: GradientColors.green,
          ),
          ContainerWidget(
            colorName: "Yellow",
            colors: GradientColors.yellow,
          ),
          ContainerWidget(
            colorName: "Orange Pink",
            colors: GradientColors.orangePink,
          ),
          ContainerWidget(
            colorName: "Aqua",
            colors: GradientColors.aqua,
          ),
          ContainerWidget(
            colorName: "Sunrise",
            colors: GradientColors.sunrise,
          ),
          ContainerWidget(
            colorName: "Cherry",
            colors: GradientColors.cherry,
          ),
          ContainerWidget(
            colorName: "Mojito",
            colors: GradientColors.mojito,
          ),
          ContainerWidget(
            colorName: "juicy Orange",
            colors: GradientColors.juicyOrange,
          ),
          ContainerWidget(
            colorName: "Mirage",
            colors: GradientColors.mirage,
          ),
          Divider(),
          Center(child: Text("More Gradient Color")),
          ContainerWidget(
            colorName: "Dark Sky Blue",
            colors: MoreGradientColors.darkSkyBlue,
          ),
          ContainerWidget(
            colorName: "Instagram",
            colors: MoreGradientColors.instagram,
          ),
          ContainerWidget(
            colorName: "JShine",
            colors: MoreGradientColors.jShine,
          ),
          ContainerWidget(
            colorName: "Cool Sky",
            colors: MoreGradientColors.coolSky,
          ),
          ContainerWidget(
            colorName: "Azure Lane",
            colors: MoreGradientColors.azureLane,
          ),
          ContainerWidget(
            colorName: "Orange Pink Teal",
            colors: MoreGradientColors.orangePinkTeal,
          ),
          Divider()
        ],
      ),
    );
  }
}

```


CHANGELOG.md
## [1.0.0] - TODO: Add release date.

- TODO: Describe initial release.

## [2.0.0-prerelease-nullsafety] - TODO: Add release date.

- TODO:Migrate To Flutter 2.0

## [2.1.0] - TODO: 29-07-2021

- New Colors Added

## [2.1.1] - TODO: 20-08-2021

-TODO:


example/test/widget_test.dart
// // This is a basic Flutter widget test.
// //
// // To perform an interaction with a widget in your test, use the WidgetTester
// // utility that Flutter provides. For example, you can send tap and scroll
// // gestures. You can also use WidgetTester to find child widgets in the widget
// // tree, read text, and verify that the values of widget properties are correct.

// import 'package:flutter/material.dart';
// import 'package:flutter_test/flutter_test.dart';

// void main() {
//   testWidgets('Counter increments smoke test', (WidgetTester tester) async {
//     // Build our app and trigger a frame.
//     await tester.pumpWidget(MyApp());

//     // Verify that our counter starts at 0.
//     expect(find.text('0'), findsOneWidget);
//     expect(find.text('1'), findsNothing);

//     // Tap the '+' icon and trigger a frame.
//     await tester.tap(find.byIcon(Icons.add));
//     await tester.pump();

//     // Verify that our counter has incremented.
//     expect(find.text('0'), findsNothing);
//     expect(find.text('1'), findsOneWidget);
//   });
// }


test/flutter_gradient_colors_test.dart
// import 'package:flutter_test/flutter_test.dart';

// import 'package:flutter_gradient_colors/flutter_gradient_colors.dart';

// void main() {
//   test('adds one to input values', () {
//     final calculator = Calculator();
//     expect(calculator.addOne(2), 3);
//     expect(calculator.addOne(-7), -6);
//     expect(calculator.addOne(0), 1);
//     expect(() => calculator.addOne(null), throwsNoSuchMethodError);
//   });
// }


example/lib/new_colors.dart
import './container_widget.dart';
import 'package:flutter/material.dart';
import 'package:flutter_gradient_colors/flutter_gradient_colors.dart';

// ignore: must_be_immutable
class NewColors extends StatelessWidget {
  List colorNames = [
    {"name": "WarmFlame", "color": GradientColors.warmFlame},
    {"name": "NightFade", "color": GradientColors.nightFade},
    {"name": "SpringWarmt", "color": GradientColors.springWarmt},
    {"name": "JuicyPeach", "color": GradientColors.juicyPeach},
    {"name": "LadyLips", "color": GradientColors.ladyLips},
    {"name": "SunnyMorning", "color": GradientColors.sunnyMorning},
    {"name": "RainyAshville", "color": GradientColors.rainyAshville},
    {"name": "FrozenDreams", "color": GradientColors.frozenDreams},
    {"name": "WinterNeva", "color": GradientColors.winterNeva},
    {"name": "DustyGrass", "color": GradientColors.dustyGrass},
    {"name": "TemptingAzure", "color": GradientColors.temptingAzure},
    {"name": "HeavyRain", "color": GradientColors.heavyRain},
    {"name": "AmyCrisp", "color": GradientColors.amyCrisp},
    {"name": "MeanFruit", "color": GradientColors.meanFruit},
    {"name": "LightBluee", "color": GradientColors.lightBluee},
    {"name": "RipeMalinka", "color": GradientColors.ripeMalinka},
    {"name": "CloudyKnoxville", "color": GradientColors.cloudyKnoxville},
    {"name": "MalibuBeach", "color": GradientColors.malibuBeach},
    {"name": "NewLife", "color": GradientColors.newLife},
    {"name": "TrueSunset", "color": GradientColors.trueSunset},
    {"name": "MorpheusDen", "color": GradientColors.morpheusDen},
    {"name": "RareWind", "color": GradientColors.rareWind},
    {"name": "WildApple", "color": GradientColors.wildApple},
    {"name": "PlumPlate", "color": GradientColors.plumPlate},
    {"name": "EverlastingSky", "color": GradientColors.everlastingSky},
    {"name": "HappyFisher", "color": GradientColors.happyFisher},
    {"name": "BlessingGet", "color": GradientColors.blessingGet},
    {"name": "SharpeyeEagle", "color": GradientColors.sharpeyeEagle},
    {"name": "LiadogaBottom", "color": GradientColors.liadogaBottom},
    {"name": "LemonGate", "color": GradientColors.lemonGate},
    {"name": "ItmeoBranding", "color": GradientColors.itmeoBranding},
    {"name": "ZeusMiracle", "color": GradientColors.zeusMiracle},
    {"name": "DeepBlue", "color": GradientColors.deepBlue},
    {"name": "HappyAcid", "color": GradientColors.happyAcid},
    {"name": "AwesomePine", "color": GradientColors.awesomePine},
    {"name": "NewYork", "color": GradientColors.newYork},
    {"name": "MixedHopes", "color": GradientColors.mixedHopes},
    {"name": "FlyHigh", "color": GradientColors.flyHigh},
    {"name": "FreshMilk", "color": GradientColors.freshMilk},
    {"name": "SnowAgain", "color": GradientColors.snowAgain},
    {"name": "FebruaryInk", "color": GradientColors.februaryInk},
    {"name": "KindSteel", "color": GradientColors.kindSteel},
    {"name": "SoftGrass", "color": GradientColors.softGrass},
    {"name": "GrownEarly", "color": GradientColors.grownEarly},
    {"name": "SharpBlues", "color": GradientColors.sharpBlues},
    {"name": "ShadyWater", "color": GradientColors.shadyWater},
    {"name": "DirtyBeauty", "color": GradientColors.dirtyBeauty},
    {"name": "GreatWhale", "color": GradientColors.greatWhale},
    {"name": "TeenNotebook", "color": GradientColors.teenNotebook},
    {"name": "PoliteRumors", "color": GradientColors.politeRumors},
    {"name": "RedSalvation", "color": GradientColors.redSalvation},
    {"name": "NightParty", "color": GradientColors.nightParty},
    {"name": "SkyGlider", "color": GradientColors.skyGlider},
    {"name": "HeavenPeach", "color": GradientColors.heavenPeach},
    {"name": "PurpleDivision", "color": GradientColors.purpleDivision},
    {"name": "AquaSplash", "color": GradientColors.aquaSplash},
    {"name": "RedLove", "color": GradientColors.redLove},
    {"name": "CleanMirror", "color": GradientColors.cleanMirror},
    {"name": "PremiumDark", "color": GradientColors.premiumDark},
    {"name": "CochitiLake", "color": GradientColors.cochitiLake},
    {"name": "SummerGames", "color": GradientColors.summerGames},
    {"name": "PassionateRed", "color": GradientColors.passionateRed},
    {"name": "MountainRock", "color": GradientColors.mountainRock},
    {"name": "DesertHump", "color": GradientColors.desertHump},
    {"name": "JungleDay", "color": GradientColors.jungleDay},
    {"name": "PhoenixStart", "color": GradientColors.phoenixStart},
    {"name": "OctoberSilence", "color": GradientColors.octoberSilence},
    {"name": "FarawayRiver", "color": GradientColors.farawayRiver},
    {"name": "AslchemistLab", "color": GradientColors.alchemistLab},
    {"name": "OverSun", "color": GradientColors.overSun},
    {"name": "MarsParty", "color": GradientColors.marsParty},
    {"name": "EternalConstance", "color": GradientColors.eternalConstance},
    {"name": "SummerGames", "color": GradientColors.japanBlush},
    {"name": "JapanBlush", "color": GradientColors.summerGames},
    {"name": "SmilingRain", "color": GradientColors.smilingRain},
    {"name": "Cloudy", "color": GradientColors.cloudy},
    {"name": "BigMango", "color": GradientColors.bigMango},
    {"name": "HealthyWater", "color": GradientColors.healthyWater},
    {"name": "Amour", "color": GradientColors.amour},
    {"name": "StrongStick", "color": GradientColors.strongStick},
    {"name": "BlackGray", "color": GradientColors.blackGray},
    {"name": "PaloAlto", "color": GradientColors.paloAlto},
    {"name": "HappyMemories", "color": GradientColors.happyMemories},
    {"name": "MidnightBloom", "color": GradientColors.midnightBloom},
    {"name": "Crystalline", "color": GradientColors.crystalline},
    {"name": "PartyBliss", "color": GradientColors.partyBliss},
    {"name": "ConfidentCloud", "color": GradientColors.confidentCloud},
    {"name": "LeCocktail", "color": GradientColors.leCocktail},
    {"name": "RiverCity", "color": GradientColors.riverCity},
    {"name": "RozenBerry", "color": GradientColors.rozenBerry},
    {"name": "ChildCare", "color": GradientColors.childCare},
    {"name": "FlyingLemon", "color": GradientColors.flyingLemon},
    {"name": "HiddenJaguar", "color": GradientColors.hiddenJaguar},
    {"name": "Nega", "color": GradientColors.nega},
    {"name": "Seashore", "color": GradientColors.seashore},
    {"name": "MarbleWall", "color": GradientColors.marbleWall},
    {"name": "CheerfulCaramel", "color": GradientColors.cheerfulCaramel},
    {"name": "NightSky", "color": GradientColors.nightSky},
    {"name": "YoungGrass", "color": GradientColors.youngGrass},
    {"name": "ColorfulPeach", "color": GradientColors.colorfulPeach},
    {"name": "GentleCare", "color": GradientColors.gentleCare},
    {"name": "PlumBath", "color": GradientColors.plumBath},
    {"name": "HappyUnicorn", "color": GradientColors.happyUnicorn},
    {"name": "GlassWater", "color": GradientColors.glassWater},
  ];

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("New Colors"),
      ),
      body: ListView(
          children: colorNames.map((colorDetails) {
        return ContainerWidget(
          colorName: "${colorDetails['name']}",
          colors: colorDetails["color"],
        );
      }).toList()),
    );
  }
}


example/lib/container_widget.dart
import 'package:flutter/material.dart';

class ContainerWidget extends StatelessWidget {
  final List<Color> colors;
  final String colorName;

  ContainerWidget({Key key, @required this.colors, @required this.colorName})
      : super(key: key);
  @override
  Widget build(BuildContext context) {
    return Center(
      child: Stack(
        children: <Widget>[
          Container(
            margin: EdgeInsets.only(top: 10.0),
            width: MediaQuery.of(context).size.width - 10,
            height: 140,
            decoration: BoxDecoration(
                gradient: LinearGradient(
              begin: Alignment.topLeft,
              end: Alignment.bottomRight,
              colors: colors,
            )),
          ),
          Positioned(
            bottom: 0,
            left: 0,
            right: 0,
            child: Container(
              height: 30.0,
              child: Center(
                child: Text(
                  "$colorName",
                  style: TextStyle(color: Colors.white, fontSize: 20.0),
                ),
              ),
              color: Color.fromRGBO(0, 0, 0, 0.3),
            ),
          )
        ],
      ),
    );
  }
}


example/lib/main.dart
import 'package:example/new_colors1.dart';

import './new_colors.dart';
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';
import 'package:flutter_gradient_colors/flutter_gradient_colors.dart';
import './container_widget.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: "Flutter Gradient Color",
      debugShowCheckedModeBanner: false,
      home: Scaffold(
        appBar: AppBar(
          title: Text("Flutter Gradient Color"),
        ),
        body: BodyPart(),
      ),
    );
  }
}

class BodyPart extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return CupertinoScrollbar(
      child: ListView(
        children: <Widget>[
          Padding(
              child: TextButton(
                style: ButtonStyle(
                    backgroundColor: MaterialStateProperty.all(Colors.indigo)),
                child: Text(
                  "New Colors",
                  style: TextStyle(color: Colors.white),
                ),
                onPressed: () {
                  Navigator.push(context,
                      MaterialPageRoute(builder: (context) => NewColors()));
                },
              ),
              padding: EdgeInsets.all(16.0)),
          Padding(
              child: TextButton(
                style: ButtonStyle(
                    backgroundColor:
                        MaterialStateProperty.all(Colors.blue[700])),
                child: Text(
                  "New Colors 29-07-2021",
                  style: TextStyle(color: Colors.white),
                ),
                onPressed: () {
                  Navigator.push(context,
                      MaterialPageRoute(builder: (context) => NewColors1()));
                },
              ),
              padding: EdgeInsets.all(16.0)),
          ContainerWidget(
            colorName: "Blue",
            colors: GradientColors.blue,
          ),
          ContainerWidget(
            colorName: "Cool Blue",
            colors: GradientColors.coolBlues,
          ),
          ContainerWidget(
            colorName: "Pink",
            colors: GradientColors.pink,
          ),
          ContainerWidget(
            colorName: "Orange",
            colors: GradientColors.orange,
          ),
          ContainerWidget(
            colorName: "Indigo",
            colors: GradientColors.indigo,
          ),
          ContainerWidget(
            colorName: "White",
            colors: GradientColors.white,
          ),
          ContainerWidget(
            colorName: "Black",
            colors: GradientColors.black,
          ),
          ContainerWidget(
            colorName: "Piggy Pink",
            colors: GradientColors.piggyPink,
          ),
          ContainerWidget(
            colorName: "Evening Sunshine",
            colors: GradientColors.eveningSunshine,
          ),
          ContainerWidget(
            colorName: "Dark Ocean",
            colors: GradientColors.darkOcean,
          ),
          ContainerWidget(
            colorName: "Grade Grey",
            colors: GradientColors.gradeGrey,
          ),
          ContainerWidget(
            colorName: "Dim Blue",
            colors: GradientColors.dimBlue,
          ),
          ContainerWidget(
            colorName: "Ver",
            colors: GradientColors.ver,
          ),
          ContainerWidget(
            colorName: "LightBlue",
            colors: GradientColors.lightBlue,
          ),
          ContainerWidget(
            colorName: "Lunada",
            colors: MoreGradientColors.lunada,
          ),
          ContainerWidget(
            colorName: "Light Green",
            colors: GradientColors.lightGreen,
          ),
          ContainerWidget(
            colorName: "Violet",
            colors: GradientColors.violet,
          ),
          ContainerWidget(
            colorName: "Light Black",
            colors: GradientColors.lightBlack,
          ),
          ContainerWidget(
            colorName: "Facebook Messenger",
            colors: GradientColors.facebookMessenger,
          ),
          ContainerWidget(
            colorName: "Winter",
            colors: GradientColors.winter,
          ),
          ContainerWidget(
            colorName: "Cloud",
            colors: GradientColors.cloud,
          ),
          ContainerWidget(
            colorName: "Grey",
            colors: GradientColors.grey,
          ),
          ContainerWidget(
            colorName: "Beautiful Green",
            colors: GradientColors.beautifulGreen,
          ),
          ContainerWidget(
            colorName: "Mango",
            colors: GradientColors.mango,
          ),
          ContainerWidget(
            colorName: "Royal Blue",
            colors: GradientColors.royalBlue,
          ),
          ContainerWidget(
            colorName: "Sky Line",
            colors: GradientColors.skyLine,
          ),
          ContainerWidget(
            colorName: "Dark Pink",
            colors: GradientColors.darkPink,
          ),
          ContainerWidget(
            colorName: "Purple Pink",
            colors: GradientColors.purplePink,
          ),
          ContainerWidget(
            colorName: "Sky Blue",
            colors: GradientColors.skyBlue,
          ),
          ContainerWidget(
            colorName: "Sea Blue",
            colors: GradientColors.seaBlue,
          ),
          ContainerWidget(
            colorName: "Hazel",
            colors: MoreGradientColors.hazel,
          ),
          ContainerWidget(
            colorName: "Noon To Dusk",
            colors: GradientColors.noontoDusk,
          ),
          ContainerWidget(
            colorName: "Red",
            colors: GradientColors.red,
          ),
          ContainerWidget(
            colorName: "Light Brown",
            colors: GradientColors.lightBrown,
          ),
          ContainerWidget(
            colorName: "Harmonic Energy",
            colors: GradientColors.harmonicEnergy,
          ),
          ContainerWidget(
            colorName: "Radish",
            colors: GradientColors.radish,
          ),
          ContainerWidget(
            colorName: "Sunny",
            colors: GradientColors.sunny,
          ),
          ContainerWidget(
            colorName: "Teal",
            colors: GradientColors.teal,
          ),
          ContainerWidget(
            colorName: "Purple",
            colors: GradientColors.purple,
          ),
          ContainerWidget(
            colorName: "Green",
            colors: GradientColors.green,
          ),
          ContainerWidget(
            colorName: "Yellow",
            colors: GradientColors.yellow,
          ),
          ContainerWidget(
            colorName: "Orange Pink",
            colors: GradientColors.orangePink,
          ),
          ContainerWidget(
            colorName: "Aqua",
            colors: GradientColors.aqua,
          ),
          ContainerWidget(
            colorName: "Sunrise",
            colors: GradientColors.sunrise,
          ),
          ContainerWidget(
            colorName: "Cherry",
            colors: GradientColors.cherry,
          ),
          ContainerWidget(
            colorName: "Mojito",
            colors: GradientColors.mojito,
          ),
          ContainerWidget(
            colorName: "juicy Orange",
            colors: GradientColors.juicyOrange,
          ),
          ContainerWidget(
            colorName: "Mirage",
            colors: GradientColors.mirage,
          ),
          Divider(),
          Center(child: Text("More Gradient Color")),
          ContainerWidget(
            colorName: "Dark Sky Blue",
            colors: MoreGradientColors.darkSkyBlue,
          ),
          ContainerWidget(
            colorName: "Instagram",
            colors: MoreGradientColors.instagram,
          ),
          ContainerWidget(
            colorName: "JShine",
            colors: MoreGradientColors.jShine,
          ),
          ContainerWidget(
            colorName: "Cool Sky",
            colors: MoreGradientColors.coolSky,
          ),
          ContainerWidget(
            colorName: "Azure Lane",
            colors: MoreGradientColors.azureLane,
          ),
          ContainerWidget(
            colorName: "Orange Pink Teal",
            colors: MoreGradientColors.orangePinkTeal,
          ),
          Divider()
        ],
      ),
    );
  }
}
